etl/steps/data/garden/state_capacity/2023-11-10/information_capacity_dataset.py

```diff
@@ -0,0 +1,149 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+import numpy as np
+import owid.catalog.processing as pr
+import pandas as pd
+from owid.catalog import Table
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("information_capacity_dataset")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["information_capacity_dataset"].reset_index()
+
+    #
+    # Process data.
+    #
+
+    # Drop country id columns
+    tb = tb.drop(columns=["ccodecow", "vdemcode"])
+
+    # Harmonize country names.
+    tb = geo.harmonize_countries(
+        df=tb, countries_file=paths.country_mapping_path, excluded_countries_file=paths.excluded_countries_path
+    )
+
+    # Reset index after dropping excluded countries
+    tb = tb.reset_index(drop=True)
+
+    # Add regional aggregations
+    tb = regional_aggregations(tb)
+
+    tb = tb.set_index(["country", "year"], verify_integrity=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def regional_aggregations(tb: Table) -> Table:
+    """
+    Add regional aggregations for some of the indicators
+    """
+
+    tb_regions = tb.copy()
+
+    # Load population data.
+    tb_pop = paths.load_dataset("population")
+    tb_pop = tb_pop["population"].reset_index()
+
+    # Merge population data.
+    tb_regions = tb_regions.merge(tb_pop[["country", "year", "population"]], how="left", on=["country", "year"])
+
+    # List of index columns
+    index_cols = ["censusgraded_ability", "ybcov_ability", "infcap_irt", "infcap_pca"]
+
+    for col in index_cols:
+        # Create new columns with the product of the index and the population
+        tb_regions[col] = tb_regions[col] * tb_regions["population"]
```

Comment:
This introduces the error
```
AssertionError: Column `censusgraded_ability` is using np.nan, but it should be using pd.NA because it has type Float64
```

How can I avoid it, @Marigold? I have never found it before, and I have proceeded similarly.

Reply:
Just for context - `np.nan` and `pd.NA` are two completely different nan values, and it's causing major headaches (ask PabloR...). The reason is appears there is that `tb_regions[col]` might have UInt8 type which is one of the types that use pd.NA and not the good old `int` type.

The fix is luckily easy - just convert it to the old "float" type (which is also used by population)
```
tb_regions[col] = tb_regions[col].astype(float) * tb_regions["population"]
```

Generally, if you keep having issues with the new types (like UInt, Float64, those starting with uppercase), just convert them to `int` or `float` as soon as possible in the pipeline.

Reply:
Thanks @Marigold! I haven't found that before. I will correct this way if I find it anywhere else